libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -113,9 +115,45 @@ bool SyntaxChecker::visit(PragmaDirective const& _pragma)
 				m_sourceUnit->annotation().experimentalFeatures.insert(feature);
 				if (!ExperimentalFeatureWithoutWarning.count(feature))
 					m_errorReporter.warning(2264_error, _pragma.location(), "Experimental features are turned on. Do not use experimental features on live deployments.");
+
+				if (feature == ExperimentalFeature::ABIEncoderV2)
+				{
+					if (m_sourceUnit->annotation().useABICoderV2.set())
+					{
+						if (!*m_sourceUnit->annotation().useABICoderV2)
+							m_errorReporter.syntaxError(
+								8273_error,
+								_pragma.location(),
+								"ABI coder V1 has already been selected through \"pragma abicoder v2\"."
```

Comment:
```suggestion
								"ABI coder V2 has already been selected through \"pragma abicoder v2\"."
```
?

Reply:
And what if `v1` was selected?

Reply:
Using `pragma abicoder v2` and `pragma experimental ABIEncoderV2` together should not be an error. Please see the negation in line 123

Reply:
> Using pragma abicoder v2 and pragma experimental ABIEncoderV2 together should not be an error.

In the same file? I guess it is moot as long as the warning is emitted. I'd make it an error but it is just not worth adding that extra code.

Reply:
> as long as the warning is emitted
But we only warn in 0.8.x, right?

Reply:
Yeah, it's fine. i still think we could just remove it entirely in 0.8.x, but perhaps we can make a decision for that when we translate external tests to 0.8.0.

Reply:
Ah I forgot this PR is on `develop`.